src/mito2/src/access.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use object_store::{util, ObjectStore};
+use snafu::ResultExt;
+
+use crate::error::{DeleteSstSnafu, Result};
+use crate::sst::file::FileId;
+
+pub type AccessLayerRef = Arc<AccessLayer>;
+
+/// Sst access layer.
+pub struct AccessLayer {
```

Comment:
I'd suggestion you to migrate all usage of `FsAccessLayer` to this `AccessLayer` and delete `AccessLayer` trait and `FsAccessLayer` before this PR so that we can avoid confision.

Reply:
TODO: At present, the name is FsAccessLayer. It may be changed to AccessLayer later.